### PR TITLE
Doc: Add and Fix docstrings for torch.util.data files 

### LINKED ISSUE #112635: Fix docstring errors in _hook_iterator.py, fetch.py, graph_settings.py, graph.py, distributed.py, gen_pyi.py, dataloader.py, signal_handling.py, collate.py, worker.py, dataset.py, sampler.py, _decorator.py, __init__.py, _typing.py, pin_memory.py
Please fix the following issues. To test locally, run: 
 ```
pydocstyle path-to-file --count
```
Run this command before and after fixing the errors. This command prints out the number of errors at the end of the output. Paste the output to your PR description with the number before and after. 
- **File**: `torch/utils/data/graph.py`, **Entity**: `traverse_dps`, **Line**: 85, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/graph.py`, **Entity**: `traverse`, **Line**: 102, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/graph.py`, **Entity**: `traverse`, **Line**: 102, **Description**: First line should end with a period (not 'n')
- **File**: `torch/utils/data/graph.py`, **Entity**: `traverse`, **Line**: 102, **Description**: First line should be in imperative mood; try rephrasing (found 'Deprecated')
- **File**: `torch/utils/data/graph_settings.py`, **Entity**: `apply_sharding`, **Line**: 50, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/graph_settings.py`, **Entity**: `apply_shuffle_settings`, **Line**: 88, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/graph_settings.py`, **Entity**: `apply_shuffle_settings`, **Line**: 88, **Description**: First line should end with a period (not 'e')
- **File**: `torch/utils/data/graph_settings.py`, **Entity**: `apply_random_seed`, **Line**: 131, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/graph_settings.py`, **Entity**: `apply_random_seed`, **Line**: 131, **Description**: First line should end with a period (not 'f')
- **File**: `torch/utils/data/dataloader.py`, **Entity**: ``, **Line**: 1, **Description**: First line should end with a period (not 'r')
- **File**: `torch/utils/data/dataloader.py`, **Entity**: `_InfiniteConstantSampler`, **Line**: 83, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/dataloader.py`, **Entity**: `DataLoader`, **Line**: 124, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/dataloader.py`, **Entity**: `DataLoader`, **Line**: 124, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/dataloader.py`, **Entity**: `DataLoader`, **Line**: 124, **Description**: First line should end with a period (not 'r')
- **File**: `torch/utils/data/dataloader.py`, **Entity**: `_MultiProcessingDataLoaderIter`, **Line**: 684, **Description**: First line should end with a period (not 'r')
- **File**: `torch/utils/data/distributed.py`, **Entity**: `set_epoch`, **Line**: 128, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/distributed.py`, **Entity**: `set_epoch`, **Line**: 128, **Description**: First line should end with a period (not 's')
- **File**: `torch/utils/data/distributed.py`, **Entity**: `set_epoch`, **Line**: 128, **Description**: First line should be in imperative mood (perhaps 'Set', not 'Sets')
- **File**: `torch/utils/data/dataset.py`, **Entity**: `IterableDataset`, **Line**: 76, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/dataset.py`, **Entity**: `TensorDataset`, **Line**: 194, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/dataset.py`, **Entity**: `StackDataset`, **Line**: 215, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/dataset.py`, **Entity**: `ConcatDataset`, **Line**: 262, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/dataset.py`, **Entity**: `ChainDataset`, **Line**: 312, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/dataset.py`, **Entity**: `Subset`, **Line**: 339, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/sampler.py`, **Entity**: `SequentialSampler`, **Line**: 99, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/sampler.py`, **Entity**: `RandomSampler`, **Line**: 117, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/sampler.py`, **Entity**: `RandomSampler`, **Line**: 117, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/sampler.py`, **Entity**: `SubsetRandomSampler`, **Line**: 175, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/sampler.py`, **Entity**: `WeightedRandomSampler`, **Line**: 196, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/_utils/signal_handling.py`, **Entity**: ``, **Line**: 1, **Description**: Use """triple double quotes""" (found """"-quotes)
- **File**: `torch/utils/data/_utils/__init__.py`, **Entity**: ``, **Line**: 1, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/_utils/__init__.py`, **Entity**: ``, **Line**: 1, **Description**: First line should end with a period (not 'y')
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: ``, **Line**: 1, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: ``, **Line**: 1, **Description**: Use """triple double quotes""" (found """"-quotes)
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: ``, **Line**: 1, **Description**: First line should end with a period (not 'o')
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `default_convert`, **Line**: 21, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `default_convert`, **Line**: 21, **Description**: Docstring is over-indented
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `default_convert`, **Line**: 21, **Description**: First line should end with a period (not ',')
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `default_convert`, **Line**: 21, **Description**: First line should be in imperative mood; try rephrasing (found 'Function')
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `collate`, **Line**: 88, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `collate`, **Line**: 88, **Description**: Docstring is over-indented
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `collate`, **Line**: 88, **Description**: First line should end with a period (not 'h')
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `default_collate`, **Line**: 205, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `default_collate`, **Line**: 205, **Description**: Docstring is over-indented
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `default_collate`, **Line**: 205, **Description**: First line should end with a period (not 'h')
- **File**: `torch/utils/data/_utils/collate.py`, **Entity**: `default_collate`, **Line**: 205, **Description**: First line should be in imperative mood; try rephrasing (found 'Function')
- **File**: `torch/utils/data/_utils/fetch.py`, **Entity**: ``, **Line**: 1, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/_utils/fetch.py`, **Entity**: ``, **Line**: 1, **Description**: Use """triple double quotes""" (found """"-quotes)
- **File**: `torch/utils/data/_utils/fetch.py`, **Entity**: ``, **Line**: 1, **Description**: First line should end with a period (not 'h')
- **File**: `torch/utils/data/_utils/pin_memory.py`, **Entity**: ``, **Line**: 1, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/_utils/pin_memory.py`, **Entity**: ``, **Line**: 1, **Description**: Use """triple double quotes""" (found """"-quotes)
- **File**: `torch/utils/data/_utils/pin_memory.py`, **Entity**: ``, **Line**: 1, **Description**: First line should end with a period (not 't')
- **File**: `torch/utils/data/_utils/worker.py`, **Entity**: ``, **Line**: 1, **Description**: Use """triple double quotes""" (found """"-quotes)
- **File**: `torch/utils/data/_utils/worker.py`, **Entity**: `get_worker_info`, **Line**: 90, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/_utils/worker.py`, **Entity**: `get_worker_info`, **Line**: 90, **Description**: First line should end with a period (not 't')
- **File**: `torch/utils/data/_utils/worker.py`, **Entity**: `get_worker_info`, **Line**: 90, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/utils/data/datapipes/_decorator.py`, **Entity**: `__init__`, **Line**: 15, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_decorator.py`, **Entity**: `__init__`, **Line**: 15, **Description**: Docstring is over-indented
- **File**: `torch/utils/data/datapipes/_decorator.py`, **Entity**: `__init__`, **Line**: 15, **Description**: First line should end with a period (not '
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `_SnapshotState`, **Line**: 9, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `_SnapshotState`, **Line**: 9, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `_simplify_obj_name`, **Line**: 21, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `_generate_input_args_string`, **Line**: 35, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `_check_iterator_valid`, **Line**: 68, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `_set_datapipe_valid_iterator_id`, **Line**: 92, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `hook_iterator`, **Line**: 110, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `hook_iterator`, **Line**: 110, **Description**: First line should end with a period (not 'f')
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `hook_iterator`, **Line**: 110, **Description**: First line should be in imperative mood; try rephrasing (found 'Hook')
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `IteratorDecorator`, **Line**: 121, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `IteratorDecorator`, **Line**: 121, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `IteratorDecorator`, **Line**: 121, **Description**: First line should end with a period (not 'o')
- **File**: `torch/utils/data/datapipes/_hook_iterator.py`, **Entity**: `_get_next`, **Line**: 138, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `issubtype`, **Line**: 52, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `_issubtype_with_constraints`, **Line**: 109, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `_DataPipeType`, **Line**: 211, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `_DataPipeType`, **Line**: 211, **Description**: First line should end with a period (not '`')
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `_DataPipeMeta`, **Line**: 249, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `_DataPipeMeta`, **Line**: 249, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `_DataPipeMeta`, **Line**: 249, **Description**: First line should end with a period (not 'd')
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `_IterDataPipeMeta`, **Line**: 341, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `_IterDataPipeMeta`, **Line**: 341, **Description**: First line should end with a period (not 's')
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `conditional_reset`, **Line**: 353, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `conditional_reset`, **Line**: 353, **Description**: First line should end with a period (not 'y')
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `reinforce_type`, **Line**: 414, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/_typing.py`, **Entity**: `reinforce_type`, **Line**: 414, **Description**: First line should end with a period (not 'd')
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `find_file_paths`, **Line**: 31, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `extract_method_name`, **Line**: 45, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `extract_method_name`, **Line**: 45, **Description**: First line should end with a period (not '"')
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `extract_method_name`, **Line**: 45, **Description**: First line should be in imperative mood (perhaps 'Extract', not 'Extracts')
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `extract_class_name`, **Line**: 59, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `extract_class_name`, **Line**: 59, **Description**: First line should end with a period (not '"')
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `extract_class_name`, **Line**: 59, **Description**: First line should be in imperative mood (perhaps 'Extract', not 'Extracts')
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `parse_datapipe_file`, **Line**: 69, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `split_outside_bracket`, **Line**: 131, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `process_signature`, **Line**: 152, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `process_signature`, **Line**: 152, **Description**: First line should end with a period (not ',')
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `get_method_definitions`, **Line**: 178, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `get_method_definitions`, **Line**: 178, **Description**: First line should end with a period (not 's')
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `main`, **Line**: 228, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/utils/data/datapipes/gen_pyi.py`, **Entity**: `main`, **Line**: 228, **Description**: First line should end with a period (not 'n')

cc @carljparker

### PR DESCRIPTION
**Pull request by guptaaryan16**
Fixes #112635 

Fix docstrings for `torch.utils.data` files.

```
Before:
> pydocstyle torch/utils/data/graph.py --count
Before: 5
After: 1

> pydocstyle torch/utils/data/graph_settings.py --count
Before: 8
After: 3

> pydocstyle torch/utils/data/dataloader.py --count
Before: 12
After: 6

> pydocstyle torch/utils/data/dataset.py --count
Before: 28
After: 23

> pydocstyle torch/utils/data/sampler.py --count
Before: 24
After: 19

> pydocstyle torch/utils/data/_utils/signal_handling.py --count
Before: 1
After: 0

> pydocstyle torch/utils/data/_utils/__init__.py --count
Before: 2
After: 0

> pydocstyle torch/utils/data/_utils/collate.py --count
Before: 20
After: 6

> pydocstyle torch/utils/data/_utils/fetch.py --count
Before: 3
After: 0

> pydocstyle torch/utils/data/_utils/pin_memory.py --count
Before: 4
After: 1 

> pydocstyle torch/utils/data/datapipes/_decorator.py --count
Before: 19
After: 16

> pydocstyle torch/utils/data/datapipes/_hook_iterator.py --count
Before: 13
After: 0

> pydocstyle torch/utils/data/datapipes/_typing.py --count
Before: 17
After: 4

> pydocstyle torch/utils/data/datapipes/gen_pyi.py --count
Before: 19
After: 4
```

cc @svekars @carljparker

### CODE REVIEW

**Review by kit1980 — CHANGES_REQUESTED** [review]

File: torch/utils/data/_utils/collate.py [hunk 1]
```diff
@@ -19,35 +18,36 @@
...
-            >>> # Example with NamedTuple
-            >>> Point = namedtuple('Point', ['x', 'y'])
-            >>> default_convert(Point(0, 0))
-            Point(x=0, y=0)
-            >>> default_convert(Point(np.array(0), np.array(0)))
-            Point(x=tensor(0), y=tensor(0))
-            >>> # Example with List
-            >>> default_convert([np.array([0, 1]), np.array([2, 3])])
-            [tensor([0, 1]), tensor([2, 3])]
+    Define function that converts each NumPy array element into a :class:`torch.Tensor`.
```
[the remark this answers is not in the document]
Comment (guptaaryan16) [reply torch/utils/data/_utils/collate.py]: Hey @kit1980 thanks for the review, fixed the changes. Can you please review it again?

**Comment by kit1980** [comment]
Please fix "Line too long" linter issues.

**Comment by guptaaryan16** [comment]
Hey @svekars @kit1980  can you approve the rest of workflows so that I can see the lint is fixed. Thanks.

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
@guptaaryan16 need to fix https://github.com/pytorch/pytorch/actions/runs/6762782751/job/18414431782?pr=112817

**Comment by guptaaryan16** [comment]
Hey @kit1980 I think now the error warning will be fixed, please approve all workflows
cc @svekars @carljparker

**Comment by guptaaryan16** [comment]
Hey @kit1980 I updated the base so that the tests can pass. Please allow the workflows again.
cc @svekars @carljparker

**Pull request closed without merging**